editor/components/post-permalink/index.js

```diff
@@ -60,7 +60,7 @@ class PostPermalink extends Component {
 		const { isCopied, isEditingPermalink } = this.state;
 		const ariaLabel = isCopied ? __( 'Permalink copied' ) : __( 'Copy the permalink' );
 
-		if ( isNew || ! previewLink ) {
+		if ( isNew ) {
```

Comment:
Can you provide context on why the necessary condition has changed? Put differently: what changed in the breaking PR (#7130) that requires adapting this line?

Reply:
So I'm a bit confused as to how it worked before and why this condition is here at all. The only time we shouldn't render the permalink editor is when the post is entirely new and hasn't been sent to the server (eg `isNew === true`).

Changing this check makes the Permalink editor behave as it did in `3.0.1`… though I'll admit the lack of tests on this had me confused as to exactly what that was, so I just manually tested it 😄

To be honest it's just an educated guess but I'd think https://github.com/WordPress/gutenberg/pull/7130/files#diff-36dbe9dade73983f5ea95f585450ff2cR1091 is the breaking change. Not sure why it's there… @aduth?

Reply:
#7130 moved the `preview_link` out of the posts endpoint into the autosaves endpoint. The `autosave` state had never previously tracked `preview_link`, so the referenced line from the pull request is related -- though not directly. Omitting the field is not the breakage; it's the fact that we no longer track `preview_link` in the objects accessed by `getEditedPostPreviewLink`. This is intentional, though apparently I did not audit the existing code to update all places where we relied on it.

In fact, the selector `getEditedPostPreviewLink` should either be removed or updated to `getAutosaveAttribute( 'preview_link' )` (as was done for `PostPreviewButton`).

That said, there's a few issues:

- We don't populate `autosave` when the post loads, so we can't rely on it existing to retrieve a preview link. Related: #7416.
- The behavior in #7130 was changed to be such that the preview button redirects its popup when a preview link becomes available, so there is a short time between an autosave starting and its completion that the preview link is unset. What are we to be rendering as a link here in the meantime?
- More generally, what is the behavior of the link when there are unsaved changes? It's odd we apply the preview link, but not the mechanics of autosaving in order to show the latest preview of the post.

So I'm inclined to wonder: Should we just set this to be the `post.link` from whatever is the most recent saved version of the post (`currentPost`)? 

There may be some additional context buried in #5756 which could shed light on why we might want a preview link here.

Reply:
Using `getCurrentPost()` means we can otherwise preserve the logic here. Thanks! 👍